Text/HTML/Parser.hs

```diff
@@ -106,7 +106,7 @@ tagOpen =
 -- TODO: This isn't right
 endTagOpen :: Parser Token
 endTagOpen = do
-    name <- takeWhile $ \c -> isAsciiUpper c || isAsciiLower c
+    name <- takeWhile $ notInClass "\x09\x0a\x0c />"
```

Comment:
I don't know anything, it just seemed like the right change and it fixed my problem, and I was hoping you'd confirm that it makes sense.

I think your code neglected to "then switch to the tag name state" (8.2.4.10).  That would mean that I still didn't get it right.  I'll try again.